Fix 2 Driver tests on Windows

gmittert opened the pull request:
Most of the driver tests currently don't work on Windows, but these two
were relatively fixable.

- specifically target OSX in order to produce generate-dSYM
- alternatively match the full path of dsymutil, which on Windows is
  given as `"C:\....\dsymutil.exe" verify-debug-info ...`

Would it be useful to have a DEBUG_NONDARWIN etc that is the same as the DEBUG check but doesn't check for generate-dSYM and doesn't pass `-target`?

jrose-apple commented:
Thanks for looking at this. I don't think this is necessarily the right change for actions.swift, which is already XFAILed on Linux and FreeBSD. I'd rather split it up into the target-agnostic tests and the target-specific tests—and then yes, a DEBUG_NONDARWIN (or DEBUG_NON_DSYM) check would make sense.

jmittert commented:
So if I'm understanding correctly, leave actions.swift as is, but also xfail on windows, then add another test which doesn't check for dsym and doesn't xfail?

I mostly just fixed these so I'd have a passing test on Windows to add in a test for https://github.com/apple/swift/pull/20140, but if I were to go through and look at a few more, is one large PR better, or is several each with a handful of tests better?

Commit:
Fix 2 Driver tests on Windows

Most of the driver tests currently don't work on Windows, but these two were relatively fixable.
- Moved the dsym checks to actions-dsym and ignore that on windows
- alternatively match the full path of dsymutil, which on Windows is
  given as "C:\....\dsymutil.exe" verify-debug-info ...

jmittert commented:
Updated to xfail on windows and split the actions test into dsym and non dsym.

jrose-apple reviewed (commented):
Sorry for the slow feedback!

jmittert commented on test/Driver/actions-dsym.swift:
```diff
@@ -0,0 +1,169 @@
+
+// XFAIL: freebsd, linux, win32
+
+// RUN: %swiftc_driver -driver-print-actions %s 2>&1 | %FileCheck %s -check-prefix=BASIC
```
Makes sense to me, I've removed them.

jrose-apple commented:
@swift-ci Please test

jrose-apple commented:
```
/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/swift/test/Driver/actions-dsym.swift:71:11: error: MULTI: expected string not found in input
// MULTI: 6: link, {1, 3, 5}, image
          ^
<stdin>:7:1: note: scanning from here
6: swift-autolink-extract, {1, 3, 5}, autolink
^
<stdin>:8:2: note: possible intended match here
7: link, {1, 3, 5, 6}, image
 ^
```
Looks like there's a -target missing somewhere.

jmittert commented:
> 
> 
> ```
> /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/swift/test/Driver/actions-dsym.swift:71:11: error: MULTI: expected string not found in input
> // MULTI: 6: link, {1, 3, 5}, image
>           ^
> <stdin>:7:1: note: scanning from here
> 6: swift-autolink-extract, {1, 3, 5}, autolink
> ^
> <stdin>:8:2: note: possible intended match here
> 7: link, {1, 3, 5, 6}, image
>  ^
> ```
> 
> Looks like there's a -target missing somewhere.

Oof yeah, that should be removed anyways since it's not being passed -g

jmittert commented:
Alright, updated and tested locally against all of Windows, MacOS, and Linux.

jrose-apple commented:
@swift-ci Please test

jrose-apple commented:
Thanks, Jason. Sorry for all the delay!
